src/include/hip/rpp_hip_common.hpp

```diff
@@ -1689,6 +1728,104 @@ __device__ __forceinline__ float rpp_hip_rng_1_inverse_transform_sampling_f32(fl
 
 // /******************** DEVICE INTERPOLATION HELPER FUNCTIONS ********************/
 
+__device__ __forceinline__ void rpp_hip_compute_bicubic_coefficient(float weight, float *coeff)
+{
+    Rpp32f x = fabsf(weight);
+    *coeff = (x >= 2) ? 0 : ((x > 1) ? (x * x * (-0.5f * x + 2.5f) - 4.0f * x + 2.0f) : (x * x * (1.5f * x - 2.5f) + 1.0f));
+}
+
+__device__ __forceinline__ void rpp_hip_compute_lanczos3_coefficient(float weight, float *coeff)
+{
+    *coeff = fabsf(weight) >= 3 ? 0.0f : (rpp_hip_math_sinc(weight) * rpp_hip_math_sinc(weight * ONE_OVER_3));
+}
+
+__device__ __forceinline__ void rpp_hip_compute_gaussian_coefficient(float weight, float *coeff)
+{
+    *coeff = expf(weight * weight * -4.0f);
+}
+
+__device__ __forceinline__ void rpp_hip_compute_triangular_coefficient(float weight, float *coeff)
+{
+    *coeff = 1 - fabsf(weight);
+    *coeff = *coeff < 0 ? 0 : *coeff;
+}
+
+__device__ __forceinline__ void rpp_hip_compute_interpolation_coefficient(RpptInterpolationType interpolationType, float weight, float *coeff)
+{
+    switch (interpolationType)
```

Comment:
@r-abishek as discussed, please see if a combination of the templated kernels and function pointers can be used to avoid using the switch cases. Thanks.

Reply:
Verified this one. It passes training, and templated switch case performs too. Merged with this PR branch now.